lib/relation-definition.js

```diff
@@ -532,37 +532,97 @@ function lookupModel(models, modelName) {
   }
 }
 
-function lookupModelTo(modelFrom, modelTo, params, singularize) {
-  if ('string' === typeof modelTo) {
+/*
+ * @param {Object} modelFrom Instance of the 'from' model
+ * @param {Object|String} modelToRef Reference to Model object to which you are
+ *  creating the relation: model instance, model name, or name of relation to model.
+ * @param {Object} params The relation params
+ * @param {Boolean} singularize Whether the modelToRef should be singularized when
+ *  looking-up modelTo
+ * @return {Object} modelTo Instance of the 'to' model
+ */
+function lookupModelTo(modelFrom, modelToRef, params, singularize) {
+  var modelTo;
+
+  if (typeof modelToRef !== 'string') {
+    // modelToRef might already be an instance of model
+    modelTo = modelToRef;
+  } else {
+    // lookup modelTo based on relation params and modelToRef
     var modelToName;
-    params.as = params.as || modelTo;
```

Comment:
this tiny line is actually quite dangerous as the intention is not only to assign `param.as` for use in this function, but also for use outside of this function.

I isolated the assignment logic into a dedicated helper, so the only purpose of `lookUpModelTo` is now to look up models!